src/shared/components/studyTagsTooltip/StudyTagsTooltip.tsx

```diff
@@ -56,31 +60,41 @@ export default class StudyTagsTooltip extends React.Component<StudyTagsTooltipPr
         return {__html: description};
     }
 
-    renderTooltip = (meta: MobxPromiseUnionType<any>) => {
+    render() {
         let overlay:any = '';
-        if (meta.isPending) {
+        if (this.studyMetadata.isPending) {
```

Comment:
instead of having a local variable "overlay" with no typing, how about just use return statements below

Reply:
I thought about that, but since the if-else doesn't cover all cases there is a problem. I don't want to replace one of the "if else" cases by "else" because then you don't see explicitly all the cases and it is more difficult to understand the code.